src/testing/TGV_scaling.cpp

```diff
@@ -0,0 +1,381 @@
+            for(unsigned int idof=0; idof<n_dofs_cell; idof++){
+                const unsigned int istate = dg->fe_collection[poly_degree].system_to_component_index(idof).first;
+                soln_at_q[iquad][istate] += dg->solution[dofs_indices1[idof]] * dg->fe_collection[poly_degree].shape_value_component(idof, qpoint, istate);
+            }
+        }
+
+        std::vector< double > convective_eigenvalues(n_quad_pts);
+        for (unsigned int isol = 0; isol < n_quad_pts; ++isol) {
+            convective_eigenvalues[isol] = pde_physics_double->max_convective_eigenvalue (soln_at_q[isol]);
+        }
+        const double max_eig = *(std::max_element(convective_eigenvalues.begin(), convective_eigenvalues.end()));
+
+        const double max_eig_mpi = dealii::Utilities::MPI::max(max_eig, mpi_communicator);
+        double cfl = 0.1 * delta_x/max_eig_mpi;
+        if(cfl < cfl_min)
+            cfl_min = cfl;
+
+    }
+    return cfl_min;
+}
+
+template <int dim, int nstate>
+int EulerTaylorGreenScaling<dim, nstate>::run_test() const
+{
+    // using Triangulation = dealii::parallel::distributed::Triangulation<dim>;
+    // std::shared_ptr<Triangulation> grid = std::make_shared<Triangulation> (mpi_communicator);
+    using Triangulation = dealii::parallel::distributed::Triangulation<dim>;
+    std::shared_ptr<Triangulation> grid = std::make_shared<Triangulation>(
+        MPI_COMM_WORLD,
+        typename dealii::Triangulation<dim>::MeshSmoothing(
+            dealii::Triangulation<dim>::smoothing_on_refinement |
+            dealii::Triangulation<dim>::smoothing_on_coarsening));
+
+    using real = double;
+
+    PHiLiP::Parameters::AllParameters all_parameters_new = *all_parameters;  
+    double left = 0.0;
+    double right = 2 * dealii::numbers::PI;
+
+    const unsigned int n_refinements = all_parameters->flow_solver_param.number_of_grid_elements_per_dimension;
+    if(all_parameters->use_curvilinear_grid){
+        //if curvilinear
+        PHiLiP::Grids::nonsymmetric_curved_grid<dim,Triangulation>(*grid, n_refinements);
+    }
+    else{
+        //if straight
+        const bool colorize = true;
+        dealii::GridGenerator::hyper_cube(*grid, left, right, colorize);
+        std::vector<dealii::GridTools::PeriodicFacePair<typename dealii::Triangulation<PHILIP_DIM>::cell_iterator> > matched_pairs;
+        dealii::GridTools::collect_periodic_faces(*grid,0,1,0,matched_pairs);
+        dealii::GridTools::collect_periodic_faces(*grid,2,3,1,matched_pairs);
+        dealii::GridTools::collect_periodic_faces(*grid,4,5,2,matched_pairs);
+        grid->add_periodicity(matched_pairs);
+        grid->refine_global(n_refinements);
+    }
+
+    std::ofstream myfile (all_parameters_new.energy_file + ".gpl"  , std::ios::trunc);
+    const unsigned int poly_degree_start= all_parameters->flow_solver_param.poly_degree;
+
+    const unsigned int poly_degree_end = 25;
```

Comment:
Is it feasible to run this high on a local computer?

Reply:
Yes, on a local computer with 4GB of RAM to be used, I could go up to p=36 with 4^3 liner grid. The test goes up to p=25 because for the curvilinear case, there is still something in high_order_grid causing the memory to spike past p=25.

Reply:
Wow! That's quite an improvement :) p=25 looked quite high from the reference of the previous capabilities.